pageserver/src/tenant/blob_io.rs

```diff
@@ -85,14 +105,28 @@ impl<'a> BlockCursor<'a> {
                 page_remain = PAGE_SZ;
             }
             let this_blk_len = min(remain, page_remain);
-            dstbuf.extend_from_slice(&buf[off..off + this_blk_len]);
+            buf_to_write.extend_from_slice(&buf[off..off + this_blk_len]);
             remain -= this_blk_len;
             off += this_blk_len;
         }
+
+        if let Some(dstbuf) = compression {
+            if compression_bits == BYTE_ZSTD {
+                let mut decoder = async_compression::tokio::write::ZstdDecoder::new(dstbuf);
```

Comment:
Should we `.reserve()` dstbuf to 8KiB here, since when we compress a value it's always 8KiB?

Reply:
I have ran the benchmarks without such reserving logic. I think it makes sense to have it, but maybe in a followup as part of a project to figure out if it improves performance. If there is a lot of compressed data that is 256 bytes large then reserving an 8 KiB buffer for it is wasteful. I would say no but it's a bit shooting in the dark without data on it.